tests/CommonExtractor.php

```diff
@@ -0,0 +1,497 @@
+            return $output;
+        }
+        // no rows found.  If incremental fetching is turned on, we need to preserve the last state
+        if ($this->incrementalFetching['column'] && isset($this->state['lastFetchedRow'])) {
+            $output = $this->state;
+        }
+        $output['rows'] = 0;
+        return $output;
+    }
+
+    private function createConnection(DatabaseParameters $parameters): \PDO
+    {
+        if ($parameters->getPort()) {
+            $dsn = sprintf(
+                "mysql:host=%s;port=%s;dbname=%s;charset=utf8",
+                $parameters->getHost(),
+                $parameters->getPort(),
+                $parameters->getDatabase()
+            );
+        } else {
+            $dsn = sprintf(
+                "mysql:host=%s;dbname=%s;charset=utf8",
+                $parameters->getHost(),
+                $parameters->getDatabase()
+            );
+        }
+        $options = [
+            \PDO::ATTR_ERRMODE => \PDO::ERRMODE_EXCEPTION,
+        ];
+        try {
+            return new \PDO($dsn, $parameters->getUser(), $parameters->getPassword(), $options);
+        } catch (\Throwable $exception) {
+            throw new UserException("Error connecting to DB: " . $exception->getMessage(), 0, $exception);
+        }
+    }
+
+    private function executeQuery(string $query, ?int $maxTries): \PDOStatement
+    {
+        $proxy = new RetryProxy($this->getLogger(), $maxTries);
+        $stmt = $proxy->call(function () use ($query) {
+            try {
+                /** @var \PDOStatement $stmt */
+                $stmt = $this->getConnection()->prepare($query);
+                $stmt->execute();
+                return $stmt;
+            } catch (\Throwable $e) {
+                try {
+                    /** @var BaseExtractorConfig $config */
+                    $config = $this->getConfig();
+                    $this->connection = $this->createConnection($config->getDbParameters());
+                } catch (\Throwable $e) {
+                };
+                throw $e;
+            }
+        });
+        return $stmt;
+    }
+
+    private function extractTable(TableParameters $table): array
+    {
+        $outputTable = $table->getOutputTable();
+
+        $this->getLogger()->info("Exporting to " . $outputTable);
+
+        $isAdvancedQuery = true;
+        if ($table->getTableDetail() && !$table->getQuery()) {
+            $isAdvancedQuery = false;
+            $query = $this->simpleQuery($table->getTableDetail(), $table->getColumns());
+        } else {
+            $query = $table->getQuery();
+        }
+        $maxTries = $table->getRetries() ?? self::DEFAULT_MAX_TRIES;
+
+        // this will retry on CsvException
+        $proxy = new RetryProxy(
+            $this->getLogger(),
+            $maxTries,
+            RetryProxy::DEFAULT_BACKOFF_INTERVAL,
+            [DeadConnectionException::class, \ErrorException::class]
+        );
+        try {
```

Comment:
Takže to je RetryProxy v RetryProxy? Když se uvnitř volá executeQuery

Reply:
no to je dusledek toho predchoziho, v novym mysql ex to tedka treba neni, ale zase nevim jestli tam tedka nevypadl ten reconnect, na to by bylo idealni si sem mergnout jeste https://github.com/keboola/db-extractor-common/pull/85
coz by melo otestovat jestli je ten retry funkcni